src/Components/Dropdown.purs

```diff
@@ -0,0 +1,171 @@
+module Ocelot.Components.Dropdown where
+
+import Prelude
+
+import Data.Array (mapWithIndex)
+import Data.Maybe (Maybe(..), maybe)
+import Effect.Aff.Class (class MonadAff)
+import Halogen as H
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Ocelot.Block.Button as Button
+import Ocelot.Block.Icon as Icon
+import Select as Select
+import Select.Utils.Setters (setContainerProps, setItemProps, setToggleProps)
+
+data Query item a
+  = HandleSelect (Select.Message (Query item) item) a
+  | Receive (Input item) a
+  | SetItems (Array item) a
+
+type State item =
+  { isOpen :: Boolean
+  , selectedItem :: Maybe item
+  , items :: Array item
+  , label :: String
+  , toString :: item -> String
+  , disabled :: Boolean
+  }
+
+type Input item =
+  { selectedItem :: Maybe item
+  , items :: Array item
+  , label :: String
+  , toString :: item -> String
+  , disabled :: Boolean
+  }
+
+data Message item = ItemSelected item
```

Comment:
Should we mimic this pattern for the future (even have a data type that represents it across components)?

https://github.com/citizennet/purescript-ocelot/blob/master/src/Components/Typeahead/Component.purs#L82

On selection, also provide the old value?


We expose a few other messages from the typeahead (though we pretty much only use `Searched` and `ItemSelected` afaik). It seems like a dropdown and typeahead should have a similar interface, but that may be more reason to cut down the typeahead messages, not expand the dropdown's messages.

https://github.com/citizennet/purescript-ocelot/blob/master/src/Components/Typeahead/Component.purs#L83